stages/org.osbuild.rpm

```diff
@@ -382,9 +367,7 @@ def main(tree, inputs, options):
     # remove temporary machine ID if it was created by us
     if machine_id_created:
         print("deleting the fake machine id")
-        machine_id_file = pathlib.Path(f"{tree}/etc/machine-id")
-        machine_id_file.unlink()
-        machine_id_file.touch(mode=0o444)
+        remove_machine_id(tree)
```

Comment:
I'm weary about moving this to utils, if we do at least the name should change as it's not actually removing the file just the content of the file (which I guess is fine but I think I would prefer something like `empty_machine_id`  or `remove_machine_id_from_file` or something.

Reply:
Or wrap the whole thing in a context that keeps track of creating/removing/emptying the file?

Reply:
I like the context idea, will try it.

Reply:
Done with a context manager, please take a look